doc/user/content/transform-data/optimization.md

```diff
@@ -109,6 +109,22 @@ CREATE INDEX pk_teachers ON teachers (id);
 
 If your query filters one or more of the join inputs by a literal equality (e.g., `WHERE t.name = 'Escalante'`), place one of those input collections first in the `FROM` clause. In particular, this can speed up [ad hoc `SELECT` queries](/sql/select/#ad-hoc-queries) by accessing collections using index lookups rather than full scans.
 
+Note that when the same input is being used in a join as well as being constrained by equalities to literals, _either_ the join _or_ the literal equalities can be sped up by an index (possibly the same index, but usually different indexes). Which of these will perform better depends on data characteristics. For example, the following query can make use of _either_ of the following two indexes, but not both at the same time:
```

Comment:
@aalexandrov, could you please check this?

Reply:
Looks good!